BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -506,17 +521,51 @@ - (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticatio
+                                                                                 [alertController dismissViewControllerAnimated:YES completion:nil];
+                                                                                 
+                                                                                 BOXLog(@"Submitting credential for authentication challenge %@", self.authenticationChallenge);
+                                                                                 [self setWebViewCanBeUsedDirectly:YES forHost:self.connection.currentRequest.URL.host];
+                                                                                 UITextField *login = alertController.textFields.firstObject;
+                                                                                 UITextField *password = alertController.textFields.lastObject;
+                                                                                 
+                                                                                 self.authenticationChallengeCredential = [NSURLCredential credentialWithUser:[login text]
+                                                                                                                                                     password:[password text]
+                                                                                                                                                  persistence:NSURLCredentialPersistenceNone];
+                                                                                 [[self.authenticationChallenge sender] useCredential:self.authenticationChallengeCredential
+                                                                                                           forAuthenticationChallenge:self.authenticationChallenge];
+                                                                                 
+                                                                                 // Clear out the authentication challenge in memory
+                                                                                 self.authenticationChallenge = nil;
+                                                                             }];
+                        
+                        [alertController addAction:cancelAction];
+                        [alertController addAction:submitAction];
+                        
+                        [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField)
+                         {
+                             textField.placeholder = NSLocalizedString(@"Username", @"Alert view text placeholder: Placeholder for where to enter user name for SSO authentication challenge");
+                         }];
+                        
+                        [alertController addTextFieldWithConfigurationHandler:^(UITextField *textField) {
+                             textField.secureTextEntry = YES;
```

Comment:
the new controller does not have a default password placeholder text. I updated this to set the password placeholder text. To note, this is in the english files, but is not translated.